lib/course_cleanup_manager.rb

```diff
@@ -0,0 +1,25 @@
+#= Class for callback cleanup methods for Courses
+class CourseCleanupManager
+  def self.cleanup_articles(course, user)
+    # find which course articles this user contributed to
+    user_articles = user.revisions
+                    .where('date >= ? AND date <= ?', course.start, course.end)
+                    .pluck(:article_id)
```

Comment:
[extract temp to query](http://www.refactoring.com/catalog/replaceTempWithQuery.html)